dspace-server-webapp/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -441,7 +442,15 @@ public ResourceSupport findRel(HttpServletRequest request, HttpServletResponse r
         throws HttpRequestMethodNotSupportedException {
         checkModelPluralForm(apiCategory, model);
         DSpaceRestRepository<RestAddressableModel, ID> repository = utils.getResourceRepository(apiCategory, model);
-        RestAddressableModel modelObject = repository.createAndReturn();
+
+        String parentCommunityString = request.getParameter("parent");
+        RestAddressableModel modelObject;
+        if (parentCommunityString != null) {
+            UUID parentCommunityUuid = UUIDUtils.fromString(parentCommunityString);
```

Comment:
postJsonInternal is a generic method. The dso type is not known at this point. It *may* have a parent. I would therefore call this parentObjectId or parentObjectUuid.